nixos/lib/test-driver/src/test_driver/driver.py

```diff
@@ -163,7 +180,34 @@ def test_script(self) -> None:
         """Run the test script"""
         with self.logger.nested("run the VM test script"):
             symbols = self.test_symbols()  # call eagerly
-            exec(self.tests, symbols, None)
+            try:
+                exec(self.tests, symbols, None)
+            except MachineError:
+                for line in traceback.format_exc().splitlines():
+                    self.logger.log_test_error(line)
+                sys.exit(1)
+            except RequestedAssertionFailed:
+                exc_type, exc, tb = sys.exc_info()
+                filtered = [
```

Comment:
```suggestion
                # Keep only stack frames mentioning the test script
                # (note: because the script is not a real file, the frame filename is `<string>`)
                filtered = [
```